### DISCUSSION TITLE
Vercel serverless functions doesn't seem to support turborepo internal packages

### DISCUSSION THREAD

**Discussion by kasper573** [vercel/turborepo#4631] [Help]
I just ran into an error after adding an internal package (https://turbo.build/repo/docs/handbook/sharing-code/internal-packages) to my api app in my turborepo. The api plugs into Vercels serverless functions using a standalone express app (https://vercel.com/guides/using-express-with-vercel#standalone-express), and it worked just fine before adding a workspace:* dependency.

As soon as I added the new dependency I'm getting this runtime error: `Cannot find module '/var/task/apps/graphql-api/node_modules/favorite-movies-db-client/src/index.ts' Did you forget to add it to "dependencies" in package.json? RequestId: 9161c7b9-5bf3-4eee-b943-42c63ab447a9 Error: Runtime exited with error: exit status 1 Runtime.ExitError`

Is this supposed to be working or am I trying to do something that's not supported?

**Comment by kennaruk** [comment]
Hi @kasper573,

I got the same issue, any solutions or workaround so far?

**Comment by kasper573** [comment]
@kennaruk Yeah I worked around it by bundling the server code and pointing the vercel api entrypoint to the bundle instead. It works all right. You can see how I did [here](https://github.com/kasper573/yas/blob/984354f3aecb9bdd287b24a5481a131bae8800c8/apps/trpc-server/api/index.ts)

**Comment by nktnet1** [comment]
Workaround for me was using a bundler like [tsdown](https://tsdown.dev) in combination with [vercel primitives](https://vercel.com/docs/build-output-api/primitives) and [prebuilt deployment](https://vercel.com/docs/cli/deploy#prebuilt).

Solution was inspired by:
- https://stackoverflow.com/a/79709576/22324694

I wrote more in-depth about the approach for my project here:
- https://github.com/nktnet1/rt-stack/discussions/21#discussioncomment-14380642
